### PR TITLE
Feature/game logic

### PR DESCRIPTION
**Pull request by MouhaDiouf**
In this pull request, we created the battleship game, with the instructions from the Odin Project. 
The player can play against a computer to win or lose the game.

He can play on a guess, by clicking on the computer board. The game will end if all the ships of either player or computer are sunk.

### CODE REVIEW

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        "SharedArrayBuffer": "readonly"
+    },
+    "parserOptions": {
+        "ecmaVersion": 2018
+    },
+
+  "globals": {
+    "__dirname" : false
+  }
+}
```
Comment (shloch) [inline .eslintrc.json]: Configure this file, same as the one from [MICROVERSE's config](https://github.com/microverseinc/linters-config/blob/master/javascript/.eslintrc.json) file.
You will surely get stickler errors to correct  😊

File: .stickler.yml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+linters:
+    flake8:
+    eslint:
+        config: './.eslintrc.json'
+    csslint:
+fixers:
+    enable: true
```
Comment (shloch) [inline .stickler.yml]: Configure this file, same as the one from [MICROVERSE's config](https://github.com/microverseinc/linters-config/blob/master/javascript/.stickler.yml) file.

File: src/gameboard.js [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+          this.board[row_x][i - 1] += " ";
+          this.board[row_x][i - 1] += `ship_${this.ships.length}`;
+
+          if (this.board[row_x - 1]) {
+            if (!this.board[row_x - 1][i - 1].match(/blank/g)) {
+              this.board[row_x - 1][i - 1] += "-blank";
+            }
+
+          }
+
```
Comment (shloch) [inline src/gameboard.js]: this your technic of implementing function bodies inside the return statement can get confusing lol. It works well, i know.
Try and organise them this manner shown in the[ ODIN PROJECT](https://www.theodinproject.com/courses/javascript/lessons/factory-functions-and-the-module-pattern)

It's easier to read, and more "industry-standard" 😊😊

(good thing they already work, you just have to organise)

File: src/player.js [hunk 4]
```diff
@@ -0,0 +1,178 @@
...
+function Player()  {
+
+  return {
+    turn: false,
+    gameEnvironment: Gameboard(),
+    placeShip: function(n, x, y){
+    const newShip =  this.gameEnvironment.createShip(n);
+      this.gameEnvironment.placeShip(newShip, x, y);
+    },
+    printBoard: function(){
```
Comment (shloch) [inline src/player.js]: this class should probably just create a player object and returns it. The rest of the functionality you added to it looks like something you should transfer to a file/module of it's own ...which is a DOM MODULE.

You mixed logic and DOM manipulation inside here

File: src/gameboard.js [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+</div>`
+        gameOverDiv.classList.add('game-over-div');
+
+        body.prepend(gameOverDiv);
+      }
+      if (document.querySelector('.restart')) {
+        const restartButton = document.querySelector('.restart');
+        restartButton.addEventListener('click', function() {
+          window.location.reload()
+        });
```
Comment (shloch) [inline src/gameboard.js]: This function also looks like something that should be in a DOM MODULE

**Comment by MouhaDiouf** [comment]
Thanks, we'll correct that.

Le mer. 5 févr. 2020 à 14:49, SHEY Louis Chia <notifications@github.com> a
écrit :

**Comment by MouhaDiouf** [comment]
Thanks a lot!

**Pull request merged**